test/pgeneralizedgaussian.jl

```diff
@@ -0,0 +1,129 @@
+        -0.0608,0.398205588436696,0.475759245161512
+        -0.0204,0.398859277127687,0.491862141925561
+        0.02,0.398862499923666,0.507978313716902
+        0.0604,0.398215241057149,0.524081470670474
+        0.1008,0.396920665528795,0.540145386578997
+        0.1412,0.394985093327439,0.556144026962162
+        0.1816,0.392417948035498,0.572051675575411
+        0.222,0.38923168040452,0.587843058334322
+        0.2624,0.385441667768147,0.603493463659977
+        0.3028,0.381066090429489,0.618978858291238
+        0.3432,0.376125786413223,0.634275997660943
+        0.3836,0.370644086205869,0.649362529994026
+        0.424,0.364646629317593,0.664217093355429
+        0.4644,0.358161164682611,0.678819404953587
+        0.5048,0.351217337070506,0.693150342089967
+        0.5452,0.343846461805699,0.707192014235536
+        0.5856,0.336081290185479,0.720927825809832
+        0.626,0.327955768047694,0.734342529336241
+        0.6664,0.319504789967085,0.747422268746731
+        0.7068,0.310763951554577,0.760154612709528
+        0.7472,0.301769302297434,0.772528577952478
+        0.7876,0.292557101311255,0.784534642652148
+        0.828,0.283163578279136,0.796164750052605
+        0.8684,0.273624701730973,0.807412302567402
+        0.9088,0.263975956669541,0.81827214670236
+        0.9492,0.254252133382257,0.828740549214512
+        0.9896,0.244487129091676,0.838815164993143
+        1.03,0.234713763897012,0.848494997211656
+        1.0704,0.224963612246627,0.857780350353393
+        1.1108,0.215266850961145,0.866672776760204
+        1.1512,0.205652124601902,0.875175017389245
+        1.1916,0.196146428753372,0.883290937490971
+        1.232,0.186775011564286,0.891025457939793
+        1.2724,0.177561293673426,0.898384482958291
+        1.3128,0.168526806435554,0.905374824976726
+        1.3532,0.159691148163201,0.912004127362079
+        1.3936,0.151071957913399,0.918280785735546
+        1.434,0.142684906177067,0.924213868574892
+        1.4744,0.13454370167416,0.929813037768908
+        1.5148,0.126660113321382,0.935088469756236
+        1.5552,0.119044006322032,0.940050777840661
+        1.5956,0.111703391230226,0.944710936230536
+        1.636,0.104644484764399,0.949080206301951
+        1.6764,0.0978717810877164,0.953170065534611
+        1.7168,0.091388132235354,0.956992139516678
+        1.7572,0.085194836349956,0.960558137361137
+        1.7976,0.0792917323859474,0.963879790822026
+        1.838,0.0736772999596982,0.966968797345075
+        1.8784,0.0683487630544158,0.969836767234444
+        1.9188,0.0633021963346127,0.97249517506594
+        1.9592,0.0585326328834392,0.97495531542798
+        1.9996,0.0540341722453821,0.977228263024933
+    ]
+
+    # CDF test.
+    for i=1:size(test, 1)
+        @test isapprox(cdf(d, test[i, 1]), test[i, 3] ; atol = 1e-6)
+    end
+
+    # PDF test.
+    for i=1:size(pdftest, 1)
+        @test isapprox(pdf(d, test[i, 1]), test[i, 2] ; atol = 1e-6)
+    end
+
+    @test mean(d) == 0
```

Comment:
shouldn't all float tests be with `≈` instead of `==`
Also, could we add tests for a less special case where not all values are 0 or 1 for safety on the computations?

Reply:
Sure, would this require a new test block to generate a new distribution with different parameters? I'm assuming we would then need to compute another array of pdf and cdf values. I've added the necessary docstring and removed the type definitions where type obviously was not needed within the methods, (thank you again for bearing with me)